pkg/controllers/job/job_controller.go

```diff
@@ -219,6 +230,25 @@ func (cc *Controller) processNextReq() bool {
 	req := obj.(apis.Request)
 	defer cc.queue.Done(req)
 
+	key := jobcache.JobKeyByReq(&req)
+
+	// prevent multi threads processing the same job simultaneously.
+	cc.jobsLock.Lock()
+	if cc.jobsMap[key] {
+		// the job is being processed by some other thread
+		cc.queue.AddRateLimited(req)
```

Comment:
if conflict after a few times, it'll take a really time to retry job. It's better to shard the request to different chanel/worker.

Reply:
Also this re-enqueue ops can lead to request processed in unexpected orders, though i  have no detailed scenario now.

Reply:
@k82cn , @hzxuzhonghu , Based on my understading sharding with single queue will not solve this issue. so i have added queueList with number of workers & did the commit. please do review the same.